wondrous-app/components/PlateRichEditor/customPlugins/PlateFloatingLink/InsertLinkModal.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState } from 'react';
+import { useFloatingLinkSelectors, floatingLinkActions, submitFloatingLink, useEditorRef } from '@udecode/plate';
+import isUrl from 'is-url';
+import Modal from '@mui/material/Modal';
+import Box from '@mui/material/Box';
+
+import { Button as CommonButton } from 'components/Common/button';
+import { LinkModal, LinkModalInput, LinkModalTitle } from './styles';
+
+const InsertLinkModal: React.FC<{
+  initialLinkText?: string;
+}> = ({ initialLinkText }) => {
+  const [linkUrl, setLinkUrl] = useState('');
+  const [linkUrlError, setLinkUrlError] = useState('Error');
+  const [linkText, setLinkText] = useState(initialLinkText);
+  const [linkTextError, setLinkTextError] = useState('Error');
+  const selectors = useFloatingLinkSelectors();
+  const editor = useEditorRef();
+  const mode = selectors.mode();
+  const isEditing = selectors.isEditing();
+  const url = selectors.url();
+  const text = selectors.text();
+  const open = (mode === 'edit' && isEditing) || mode === 'insert';
+
+  useEffect(() => {
+    setLinkUrl(url);
+    setLinkText(text);
+  }, [mode, url, text]);
+
+  useEffect(() => {
+    const onEscPress = (e) => {
+      if (e.key === 'Escape') {
+        floatingLinkActions.hide();
+      }
+    };
+
+    document.addEventListener('keydown', onEscPress);
```

Comment:
optional but we can use react-hotkeys-hook here? not a big deal, more for consistency